### PR TITLE
Update k8s libraries to 0.19.2

### LINKED ISSUE #1293: Update k8s libraries to 0.19.x
/kind feature

**Describe the solution you'd like**
It would be great to update [our k8s dependencies](https://github.com/kubeflow/kfserving/blob/master/go.mod#L49-L74) to 0.19. It will provide compatibility with 0.19 k8s versions and allow projects using kfserving internally to have an easier migration to 0.19. 

**Anything else you would like to add:**
The last time the libraries were updated was in [this PR](https://github.com/kubeflow/kfserving/pull/1144/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R34).

### PR DESCRIPTION
**Pull request by ivan-valkov**
**What this PR does / why we need it**:
This PR updates [our k8s dependencies](https://github.com/kubeflow/kfserving/blob/master/go.mod#L49-L74) to 0.19. It will provide compatibility with 0.19 k8s versions and allow projects using kfserving internally to have an easier migration to 0.19. 
Only a small modification was needed to be able to use 0.19 - [updating the copy of podSpec we have](https://github.com/kubeflow/kfserving/compare/master...ivan-valkov:update-k8s-0.19.2?expand=1#diff-4bc42cc69a5f3db07b4b385408e396486044295aaa11b94fce4e71174f67ecafR245-R251).

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1293

**Special notes for your reviewer**:

I have tested this change by running `make test`. All the formatting changes were made from running `make test`.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: docs/samples/v1alpha2/custom/grpc-server/client/client.go [hunk 1]
```diff
@@ -8,7 +8,7 @@ import (
 	"log"
 	"time"
 
-	pb "helloworld/proto"
+	pb "github.com/kubeflow/kfserving/docs/samples/v1alpha2/custom/grpc-server/proto"
```
Comment (ivan-valkov) [inline docs/samples/v1alpha2/custom/grpc-server/client/client.go:11]: had to update this to make go mod happy

**Review by adriangonz — COMMENTED** [review]
Hey @ivan-valkov , changes look good! :+1: Do you foresee any backwards compatibility issues running in older clusters?

**Comment by ivan-valkov** [comment]
> Hey @ivan-valkov , changes look good!  Do you foresee any backwards compatibility issues running in older clusters?

Good question @adriangonz , I do not expect there to be any issues, but let me test some basic workflows on a k8s 1.15 cluster and I will get back to you.

**Comment by ivan-valkov** [comment]
> > Hey @ivan-valkov , changes look good!  Do you foresee any backwards compatibility issues running in older clusters?
> 
> Good question @adriangonz , I do not expect there to be any issues, but let me test some basic workflows on a k8s 1.15 cluster and I will get back to you.

I tested on a kind cluster and creation and deletion work fine. I think there won't be backwards compatibility issues with updating these libs @adriangonz

**Comment by yuzisun** [comment]
@ivan-valkov  can you help sign the cla?

File: go.mod [hunk 2]
```diff
@@ -18,58 +17,55 @@ require (
...
+	k8s.io/kube-proxy => k8s.io/kube-proxy v0.19.2
+	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.19.2
+	k8s.io/kubectl => k8s.io/kubectl v0.19.2
+	k8s.io/kubelet => k8s.io/kubelet v0.19.2
+	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.19.2
+	k8s.io/metrics => k8s.io/metrics v0.19.2
+	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.19.2
 	k8s.io/test-infra => k8s.io/test-infra v0.0.0-20200803112140-d8aa4e063646
 	k8s.io/utils => k8s.io/utils v0.0.0-20200324210504-a9aa75ae1b89
 	// based on https://github.com/openshift/cluster-ingress-operator/pull/429/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R34
```
Comment (yuzisun) [inline go.mod:71]: Can you help remove the following `controller-runtime` pin and bump up to 0.7.0? I think we no longer need it once move to 1.19.2.

Comment (ivan-valkov) [reply go.mod:71]: I gave it a try but it turns out we use a bunch of deprecated packages from `controller-runtime` that are deleted in `0.7.0`. These are `pkg/runtime/log` and `pkg/runtime/signals`. The issue mentioned in this comment (https://github.com/kubernetes-sigs/controller-runtime/issues/1033) is fixed only from 0.7.0 onwards. I created https://github.com/kubeflow/kfserving/issues/1307 to track the migration away from the deprecated packages and the update of this dependency.

**Comment by ivan-valkov** [comment]
> @ivan-valkov can you help sign the cla?

@yuzisun I believe I have signed it now, is there a way to check? Can I rerun the `cla/google` check somehow?

**Comment by ivan-valkov** [comment]
@googlebot I signed it!

**Comment by PatrickXYS** [comment]
/test kubeflow-kfserving-presubmit

Test to see if E2E test works fine without `Unauthorized` issue

**Comment by ivan-valkov** [comment]
Hey, hope you had a good weekend. I resolved a conflict with master and tests seem to be passing. Is there anything else we need to get this one merged @yuzisun @PatrickXYS

**Comment by PatrickXYS** [comment]
```
FAILED predictor/test_sklearn.py::test_sklearn_v2_kfserving - KeyError: 'outp...
```

Ref: http://86308603-argo-argo-5ce9-1162466691.us-west-2.elb.amazonaws.com/workflows/kubeflow-test-infra/kubeflow-kfserving-presubmit-e2e-1305-7b34a42-0256-a794?tab=workflow&nodeId=kubeflow-kfserving-presubmit-e2e-1305-7b34a42-0256-a794-4239156108&sidePanel=logs%3Akubeflow-kfserving-presubmit-e2e-1305-7b34a42-0256-a794-4239156108%3Amain

**Comment by adriangonz** [comment]
/test kubeflow-kfserving-presubmit


Testing to see if the 401 occurs again.

**Comment by PatrickXYS** [comment]
```
 Failed: cannot unmarshall spec: cannot convert int64 to string
```

Please rebase master branch to make sure it updates test code

**Comment by pugangxa** [comment]
Still ``` Failed: cannot unmarshall spec: cannot convert int64 to string```, is there something wrong with the agro workflow?

**Comment by ivan-valkov** [comment]
> ```
>  Failed: cannot unmarshall spec: cannot convert int64 to string
> ```
> 
> Please rebase master branch to make sure it updates test code

just rebased on master. Let's retest this :)

**Comment by ivan-valkov** [comment]
`kubeflow-kfserving-presubmit` now passes after the rabase - http://prow.kubeflow-testing.com/view/s3/aws-kubeflow-jenkins/pr-logs/pull/kubeflow_kfserving/1305/kubeflow-kfserving-presubmit/1356545934145622016

**Pull request merged**